nx-os/poap/poap.py

```diff
@@ -275,14 +275,15 @@ def abort_cleanup_exit () :
 import signal
 import string
 
-# Set CDP variables for location option
-# Will be used by set_config_file_src_location() function
-poap_log("INFO: show cdp neighbors interface %s" % cdp_interface)
-a = clid("show cdp neighbors interface %s" % cdp_interface)
-b = json.loads(a)
-cdpnei_switchName = str(b['TABLE_cdp_neighbor_brief_info']['ROW_cdp_neighbor_brief_info']['device_id'])
-cdpnei_intfName = str(b['TABLE_cdp_neighbor_brief_info']['ROW_cdp_neighbor_brief_info']['port_id'])
-cdpnei_intfName = string.replace(cdpnei_intfName, "/", "_")
+if config_file_type == "location" and cdp_interface is not None:
```

Comment:
The check for 'cdp_interface is not None' is redundant at this point in the execution path (since the script has already exited if it is None at line 224-226), but I'm just nitpicking at this point.